metadata/src/utils/validation.rs

```diff
@@ -405,15 +405,28 @@ pub fn get_custom_metadata_hash(custom_metadata: &CustomMetadata) -> [u8; HASH_L
 }
 
 /// Obtain the hash of some custom value in the metadata including it's name/key.
+///
+/// If the `custom_value` has a type id that is not present in the metadata,
+/// only the name and bytes are used for hashing.
 pub fn get_custom_value_hash(
     custom_value: &CustomValueMetadata,
     cache: &mut HashMap<u32, CachedHash>,
 ) -> [u8; HASH_LEN] {
-    concat_and_hash3(
-        &hash(custom_value.name.as_bytes()),
-        &get_type_hash(custom_value.types, custom_value.type_id(), cache),
-        &hash(custom_value.bytes()),
-    )
+    let name_hash = hash(custom_value.name.as_bytes());
+    let bytes_hash =hash(custom_value.bytes());
```

Comment:
The goal of this validation stuff is to make sure that the code generated from some metadata is still compatible with some node (and the node metadata) so that it can still be used without issue.

With that in mind, we don't actually need to check that `custom_value.bytes()` is identical; it doesn't matter if the bytes are different so long as they still decode OK into the generated type that the codegen is expecting. So let's remove the hashing of bytes. Otherwise looks good offhand (though this stuff always needs some thought and I obv missed this last time :))